src/v4l/v4lt.rs

```diff
@@ -0,0 +1,166 @@
+use v4l::buffer::Type;
+use v4l::io::traits::OutputStream;
+use v4l::prelude::*;
+use v4l::video::Output;
+use v4l::{Format, FourCC};
+use neolink_core::{
+    bc_protocol::{StreamOutput, StreamOutputError},
+    bcmedia::model::*,
+};
+
+type Result<T> = std::result::Result<T, ()>;
+
+pub(crate) struct V4lDevice {
+    device: Device,
+}
+
+pub(crate) struct V4ltOutputs<'a> {
+    device: Device,
+    stream: &'a mut MmapStream<'a>,
```

Comment:
First I think we need to remove the `&mut` at the front. You seem to have ownership of it at this point. 

However I suspect that there might be another issue in that the `MmapStream` has a lifetime parameter. This usually means that there is a reference in it to that points to data else where. The compiler needs to be certain that the lifetime of `'a` does not die. Which it cannot be certain of because of it running on a different thread. This will be an issue.